iceoryx_binding_c/test/moduletests/test_types.cpp

```diff
@@ -78,5 +80,10 @@ TEST(iox_types_test, ClientStorageSizeFits)
     EXPECT_THAT(alignof(UntypedClient), Le(alignof(iox_client_storage_t)));
 }
 
-
+TEST(iox_types_test, ServiceDiscoveryStorageSizeFits)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "8fd3af84-ca73-4e38-a061-5e4638f48b77");
+    EXPECT_THAT(sizeof(ServiceDiscovery), Le(sizeof(iox_service_discovery_storage_t)));
```

Comment:
We may waste a lot of storage like this (we only check <=). Once https://github.com/eclipse-iceoryx/iceoryx/pull/1151 is merged the size should shrink considerably.

Reply:
Then you have to update `types.h` :)